### PR TITLE
refactor first handler section

### PR DESCRIPTION
**Pull request by scottrfrancis**
*Issue #, if available:*

*Description of changes:*
start on a test suite and a small refactoring of some common bugs.
- use get instead of key access for default
- make globals clearly global
- eliminate switch / if/ else blocks -- bug prone and can be hard to follow
  -- CF: https://blog.cleancoder.com/uncle-bob/2021/03/06/ifElseSwitch.html

+ add header

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (schuettc) [inline .vscode/settings.json:1]: Why is this file included?

File: src/lambda_function.py [hunk 2]
```diff
@@ -1,61 +1,82 @@
...
 
 
 # Set LogLevel using environment variable, fallback to INFO if not present
 logger = logging.getLogger()
-try:
-    log_level = os.environ['LogLevel']
-    if log_level not in ['INFO', 'DEBUG']:
-        log_level = 'INFO'
-except:
+log_level = os.getenv('LogLevel', 'INFO')
```
Comment (schuettc) [inline src/lambda_function.py:25]: Why was the try/except removed?

File: src/lambda_function.py [hunk 3]
```diff
@@ -1,61 +1,82 @@
...
+
+
+# handler mapping table -- calls the method 'value' for the event 'key'
+handlers = {
+    'NEW_INBOUND_CALL': lambda call_id, event, participants: new_call_handler(call_id),
+    'RINGING': lambda call_id, event, participants: response(),
+    'ACTION_SUCCESSFUL': lambda call_id, event, participants: action_success_handler(call_id, event),
+    'DIGITS_RECEIVED': lambda call_id, event, participants: control_voicefocus(call_id, event),
+    'HANGUP': lambda call_id, event, participants: hangup(participants),
+    # 'ACTION_FAILED': lambda call_id, event, participants: (
```
Comment (schuettc) [inline src/lambda_function.py:43]: Testing Comments should be removed.

File: src/lambda_function.py [hunk 4]
```diff
@@ -82,21 +103,24 @@ def response(*actions):
 # For new incoming calls, play greeting and collect digits of destination number.
 # Regex for digits entered allows US calling, except for premium rate numbers
 def new_call_handler(call_id):
-        logger.info('SEND {} {}'.format(log_prefix, 'Sending PlayAndGetDigits action to get Destination Number'))
-        return response(pause_action(call_id), play_and_get_digits_action(call_id, '^(?!1900)1[0-9][0-9][0-9](\d{7})$', 'welcome_vf_demo.wav', 'invalid_entry.wav'))
+    logger.info('SEND {} {}'.format(
```
Comment (schuettc) [inline src/lambda_function.py:106]: Is this just a linter change?  Or were carriage returns added intentionally?

File: src/lambda_function.py [hunk 5]
```diff
@@ -82,21 +103,24 @@ def response(*actions):
...
+        log_prefix, 'Sending PlayAndGetDigits action to get Destination Number'))
+    return response(pause_action(call_id), play_and_get_digits_action(call_id, '^(?!1900)1[0-9][0-9][0-9](\d{7})$', 'welcome_vf_demo.wav', 'invalid_entry.wav'))
```
Comment (schuettc) [inline src/lambda_function.py:108]: It may be worth expanding this regex and moving it to a variable.

File: src/lambda_function.py (hunk 2 above)
Comment (scottrfrancis) [reply src/lambda_function.py:25]: not needed with 'getenv' -- which defaults to 'INFO' if the key isn't found.  Accomplishes the same result with more economy

File: src/lambda_function.py (hunk 5 above)
Comment (scottrfrancis) [reply src/lambda_function.py:108]: good idea

File: .vscode/settings.json (hunk 1 above)
Comment (scottrfrancis) [reply .vscode/settings.json:1]: including it documents (via the IDE) the linter settings and setup to run the unit tests in the IDE. As a general rule, I wouldn't include IDE specific settings, but given the broad usage of VS Code and the utility of including the file, I wanted to see what you all thought

File: src/lambda_function.py (hunk 3 above)
Comment (scottrfrancis) [reply src/lambda_function.py:43]: agreed.  this PR should NOT really be merged... i wanted to stimulate discussion on the approach before completing the refactor.

File: src/lambda_function.py (hunk 4 above)
Comment (scottrfrancis) [reply src/lambda_function.py:106]: linter
